Thorfile

```diff
@@ -1,16 +1,58 @@
+require 'yaml'
+require 'fileutils'
+require 'tmpdir'
+require 'digest'
 require 'thor/group'
 
 module Middleman
   class Generator < ::Thor::Group
     include ::Thor::Actions
 
-    source_root File.expand_path(File.dirname(__FILE__))
+    TEMPLATE_VERSION_FILE = '.template_version'.freeze
+
+    class_option 'template',
+                 aliases: '-T',
+                 default: 'alphagov/tech-docs-template'
+
+    source_root __dir__
 
     def detect_if_first_time_install
-      if option_set?('FIRST_TIME')
-        @first_time = parse_boolean('FIRST_TIME')
-      else
-        @first_time = yes?('Are you creating a completely new documentation project?')
+      @first_time = option_set?('FIRST_TIME') || !File.exist?('config.rb')
+    end
+
+    def clone_existing_version
+      return if @first_time || !File.exist?(TEMPLATE_VERSION_FILE)
+      template = YAML.load_file(TEMPLATE_VERSION_FILE)
```

Comment:
At the moment it will raise an exception and crash out. I don't know how much we want to wrap users up in cotton wool – a friendly message will be useful to users, but they wouldn't understand what to do with the version file anyway. I'd perhaps rather it explodes violently so they get a full stack trace which they can supply to someone technical – non-technical users aren't going to be able to get the file back into a useful state anyway.